concepts-wip/strings/about.md

```diff
@@ -0,0 +1,178 @@
+In Crystal is a String a sequence of unicode characters.
+Strings are immutable, meaning they cannot be changed once created.
+This means that every time you do a modification on a string a new string is created, thereby the original string is not modified.
+
+To define a string you can use double quotes `"`.
+
+```crystal
+"Hello World"
+```
+
+## Concatenation
+
+To concatenate two strings you can use the `+` operator.
+The `+` operator will return a new string that is the combination of the two strings given.
+
+```crystal
+hello = "Hello"
+hello + " World"
+# => "Hello World"
+```
+
+When wanting to concat a string with a non-string type so can you use the method `to_s` to convert the type to a string.
+
+```crystal
+"Hello " + 1.to_s
+# => "Hello 1"
+```
+
+## Conversation between Strings and numbers
+
+A string can be converted to a number using the `to_i` method, which will return an integer.
+Alternatively, the `to_f` method can be used to return a floating point number.
+If the string can't be converted to a number then an ArgumentError will be raised.
+
+```crystal
+"1".to_i
+# => 1
+
+"1.0".to_f
+# => 1.0
+
+"Hello".to_i
+# => ArgumentError: Invalid integer: Hello
+```
+
+An integer or floating point number can be converted to a string using the `to_s` method.
+
+```crystal
+1.to_s
+# => "1"
+
+1.0.to_s
+# => "1.0"
+```
+
+## Interpolation
+
+Interpolation is a convenient way to combine strings and embeds expressions in strings.
+To interpolate a string you can use the `#` character followed by curly braces `{}` with the expression inside the braces.
+
+```crystal
+name = "World"
+"Hello #{name}!"
+# => "Hello World!"
+```
+
+Crystal will automatically convert the result of the expression to a string.
+
+```crystal
+"Hello #{1 + 1}!"
+# => "Hello 2!"
+```
+
+## Size
+
+When you need to know the number of characters in a string you can use the `size` method, which will return the length of the string as an integer.
+The size of a string is a stored property of the string, so it doesn't have to calculate the size every time you call the method, making it very fast.
+
+```crystal
+"Hello World".size
+# => 11
+```
+
+## Indexing
+
+Indexing is when you want to get a specific character from a string.
+To get a character from a string you can use the `[]` method with the index of the character you want to get.
+In crystal is the first character in a string at index 0.
+
+```crystal
+"Hello World"[0]
+# => 'H'
+
+"Hello World"[6]
+# => 'W'
+```
+
+To get a character from the end of a string you can take the length of the string minus one to get the index of the last character and to get the second last character you can take the length of the string minus two and so on.
+There is a shortcut for this and that is to simply use a negative index, where the last character is at index -1, the second last character is at index -2, and so on.
+
+```crystal
+name = "Hello World"
+name[name.size - 1]
+# => 'd'
+
+name[-1]
+# => 'd'
+```
+
```

Comment:
In the About document, you might want to talk about other ways to index a string: ranges, regex, substring
```crystal
icr:1> s = "Hello World"
 => "Hello World"
icr:2> s[1]
 => 'e'
icr:3> s[1..4]
 => "ello"
icr:4> s[/e.*o/]
 => "ello Wo"
icr:5> s["nope"]
Unhandled exception: Nil assertion failed (NilAssertionError)
```

Reply:
This is something I have thought about in how to deal with this. Since a range is actually a datatype in Crystal and you can do a range over more than just numbers. So personally I thought it would make more sense to introduce that later when chars have been explained. Ranges are a bit complicated in how they should be introduced since it deals with numbers, strings, chars and arrays. 

I mean we could do a short introduction and then do a full introduction in a later concept. What do you think?

Substring, that I think could make sense to introduce here so I think that is a good idea.

Regex, I believe deserves its own concept, and I think perhaps a mention could happen but not a deeper dive with examples. Do you think that makes sense?

Reply:
If it goes into About, you don't need to put it in the exercise. It can be a "teaser" for later concepts. Range and Regex are definitely their own concepts. So, yes, I agree.